src/Extensions/SwiftypeSiteTreeCrawlerExtension.php

```diff
@@ -171,4 +200,60 @@ protected function getLogger(): LoggerInterface
     {
         return Injector::inst()->get(LoggerInterface::class);
     }
+
+
+    public function setUrlsToCrawl(array $urls) {
+        $this->urlsToCrawl = $urls;
+    }
+
+    public function getUrlsToCrawl(): array
+    {
+        return $this->urlsToCrawl;
+    }
+
+    private function getOwnerKey(): string
+    {
+        $owner = $this->getOwner();
+        $key = $owner->ClassName . $owner->ID;
+        return $key;
+    }
+
+    /**
+     * collate Urls to crawl
+     *
+     * @return void
+     */
+    public function collateUrls(): void
+    {
+        $urls = $this->getUrlsToCrawl();
+        $this->withVersionContext(function() use ($urls) {
+            /** @var SiteTree $owner */
+            $owner = $this->getOwner();
+            $key = $this->getOwnerKey();
```

Comment:
Do we need to store these with a `$key`? I'm just trying to think of when you'd ever **not** want any/all URLs.

Reply:
see above